From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pavel Tumakaev <p.tumakaev@omprussia.ru>
Date: Tue, 26 Jan 2021 00:04:24 +0300
Subject: [PATCH] [sailfishos][gecko] Disable unaligned FP access emulation on
 ARM for WebAssembly. Fixes JB#53095

On ARM Linux, including Android, unaligned FP accesses that were not
flagged as unaligned will tend to trap (with SIGBUS) and will need to be
emulated. This emulation can only be performed if the system header files
provide access to the FP registers.  In particular, <sys/user.h> must have
definitions of `struct user_vfp` and `struct user_vfp_exc`, as it does on
Android. Otherwise, emulation should be disabled to prevent compilation
errors.

See upstream commit: 6ce9a44465230fe42a2d5597d72b514284809015

Signed-off-by: Pavel Tumakaev <p.tumakaev@omprussia.ru>
---
 js/src/wasm/WasmSignalHandlers.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/js/src/wasm/WasmSignalHandlers.cpp b/js/src/wasm/WasmSignalHandlers.cpp
index 78d154408435..a1a049894913 100644
--- a/js/src/wasm/WasmSignalHandlers.cpp
+++ b/js/src/wasm/WasmSignalHandlers.cpp
@@ -244,7 +244,7 @@ using mozilla::DebugOnly;
 // If you run into compile problems on a tier-3 platform, you can disable the
 // emulation here.
 
-#if defined(__linux__) && defined(__arm__)
+#if 0
 #  define WASM_EMULATE_ARM_UNALIGNED_FP_ACCESS
 #endif
 
-- 
2.31.1

